.github/workflows/cmake-multi-platform.yml

```diff
@@ -0,0 +1,94 @@
+# This starter workflow is for a CMake project running on multiple platforms. There is a different starter workflow if you just want a single platform.
+# See: https://github.com/actions/starter-workflows/blob/main/ci/cmake-single-platform.yml
+name: CMake on multiple platforms
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+env:
+  # global build variables
+  SFML_DIR: "C:/SFML/SFML-2.6.1"
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      # Set fail-fast to false to ensure that feedback is delivered for all matrix combinations. Consider changing this to true when your workflow is stable.
+      fail-fast: false
+      
+
+      # Set up a matrix to run the following 3 configurations:
+      # 1. <Windows, Release, latest MSVC compiler toolchain on the default runner image, default generator>
+      # 2. <Linux, Release, latest GCC compiler toolchain on the default runner image, default generator>
+      # 3. <Linux, Release, latest Clang compiler toolchain on the default runner image, default generator>
+      #
+      # To add more build types (Release, Debug, RelWithDebInfo, etc.) customize the build_type list.
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        build_type: [Release]
+        c_compiler: [gcc, cl]
+        include:
+          - os: windows-latest
+            c_compiler: cl
+            cpp_compiler: cl
+          - os: ubuntu-latest
+            c_compiler: gcc
+            cpp_compiler: g++
+        exclude:
+          - os: windows-latest
+            c_compiler: gcc
+          - os: windows-latest
+            c_compiler: clang
+          - os: ubuntu-latest
+            c_compiler: cl
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set reusable strings
+      # Turn repeated input strings (such as the build output directory) into step outputs. These step outputs can be used throughout the workflow file.
+      id: strings
+      shell: bash
+      run: |
+        echo "build-output-dir=${{ github.workspace }}/build" >> "$GITHUB_OUTPUT"
+  
+    - name: Download SFML for Ubuntu
+      if:  matrix.os == 'ubuntu-latest'
+      run: sudo apt-get install -y libsfml-dev
+    
+    - name: Download SFML for Windows
+      if:  matrix.os == 'windows-latest'
+      shell: cmd
+      run: |
+        mkdir "C:/SFML" && curl -fSL "https://www.sfml-dev.org/files/SFML-2.6.1-windows-vc17-64-bit.zip" -o "C:/SFML/SFML-2.6.1-windows-vc17-64-bit.zip"
+
+    - name: Unzip SFML for Windows
+      if:  matrix.os == 'windows-latest'
+      shell: cmd
+      run: tar -xvf "C:/SFML/SFML-2.6.1-windows-vc17-64-bit.zip" -C "C:/SFML/"
```

Comment:
These 2 steps can be combined in a single "Set Up SFML (Windows)" step

Also, I noticed you are not making use of that `SFML_DIR` env var defined at the TOP. I think you should either use it or lose it

Personally, I would simply save the SFML version, either `2.6.1` or `SFML-2.6.1` as a variable and use that, such that updating SFML is easy (use it both in the download URL as well as when building paths.

Reply:
This `SFML_DIR` is actually used by cmake, instead of `SFML_HOME`. So i will remove SFML_HOME

Reply:
I have done this already. Combined 2 steps into one. Also , SFML_HOME (root directory) is  used to copy the DLL for audio on Windows only (see CmakeLists.txt). SFML_DIR is actually `SFML_HOME/lib/cmake` on Windows. Without this, Cmake on windows cannot locate sfml cmake scripts files. On Linux and MacOS this does not matter.